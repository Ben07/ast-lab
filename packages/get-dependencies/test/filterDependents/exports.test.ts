// @ts-ignore
import { enumerateArgsTestFunction, configArgs } from 'lazy-jest';
import { filterDependents } from './utils';

enumerateArgsTestFunction(
  filterDependents.bind(null, 'exports'),
  configArgs()
  .arg('sources', [
    ['./exportNamed.js', './exportDefault.js', 'exportAllFrom.js'],
    ['./exportNamed.js', './exportDefault.js',],
    ['./exportNamed.js', './exportDefault.js', 'exportDefaultAliasFrom.js'],
    ['./exportNamed.js', './exportDefault.js', 'exportNamedFrom.js'],
    ['./exportNamed.js', './exportDefault.js', 'noExports.js']
  ])
  .arg('targets', [
    { './exportNamed': ['a'] },
    { './exportDefault': ['default'] }
  ]),
  'exports from'
);