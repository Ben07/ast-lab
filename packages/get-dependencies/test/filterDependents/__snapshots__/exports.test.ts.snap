// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exports from - (["sources", "targets"]) - [["exportNamed.js","exportDefault.js","exportAllFrom.js"],{"./exportDefault":["default"]}] 1`] = `
Array [
  "/exports/exportDefault.js",
]
`;

exports[`exports from - (["sources", "targets"]) - [["exportNamed.js","exportDefault.js","exportAllFrom.js"],{"./exportNamed":["a"]}] 1`] = `
Array [
  "/exports/exportNamed.js",
  "/exports/exportAllFrom.js",
]
`;

exports[`exports from - (["sources", "targets"]) - [["exportNamed.js","exportDefault.js","exportDefaultAliasFrom.js"],{"./exportDefault":["default"]}] 1`] = `
Array [
  "/exports/exportDefault.js",
  "/exports/exportDefaultAliasFrom.js",
]
`;

exports[`exports from - (["sources", "targets"]) - [["exportNamed.js","exportDefault.js","exportDefaultAliasFrom.js"],{"./exportNamed":["a"]}] 1`] = `
Array [
  "/exports/exportNamed.js",
]
`;

exports[`exports from - (["sources", "targets"]) - [["exportNamed.js","exportDefault.js","exportNamedFrom.js"],{"./exportDefault":["default"]}] 1`] = `
Array [
  "/exports/exportDefault.js",
]
`;

exports[`exports from - (["sources", "targets"]) - [["exportNamed.js","exportDefault.js","exportNamedFrom.js"],{"./exportNamed":["a"]}] 1`] = `
Array [
  "/exports/exportNamed.js",
  "/exports/exportNamedFrom.js",
]
`;

exports[`exports from - (["sources", "targets"]) - [["exportNamed.js","exportDefault.js","noExports.js"],{"./exportDefault":["default"]}] 1`] = `
Array [
  "/exports/exportDefault.js",
]
`;

exports[`exports from - (["sources", "targets"]) - [["exportNamed.js","exportDefault.js","noExports.js"],{"./exportNamed":["a"]}] 1`] = `
Array [
  "/exports/exportNamed.js",
]
`;

exports[`exports from - (["sources", "targets"]) - [["exportNamed.js","exportDefault.js"],{"./exportDefault":["default"]}] 1`] = `
Array [
  "/exports/exportDefault.js",
]
`;

exports[`exports from - (["sources", "targets"]) - [["exportNamed.js","exportDefault.js"],{"./exportNamed":["a"]}] 1`] = `
Array [
  "/exports/exportNamed.js",
]
`;
